src/PrestaShopBundle/Controller/Admin/Sell/Catalog/FeatureController.php

```diff
@@ -26,163 +26,155 @@
-    {
-        if (!$this->isFeatureEnabled()) {
-            return $this->render('@PrestaShop/Admin/Sell/Catalog/Features/create.html.twig', [
-                'showDisabledFeatureWarning' => true,
-            ]);
-        }
-
-        $featureFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.feature_form_builder');
-        $featureFormHandler = $this->get('prestashop.core.form.identifiable_object.handler.feature_form_handler');
-
-        $featureForm = $featureFormBuilder->getForm();
-        $featureForm->handleRequest($request);
+    use AddActionTrait;
+    use SmartyTrait;
 
-        try {
-            $handlerResult = $featureFormHandler->handle($featureForm);
+    public const DEFAULT_THEME = 'default';
+    public const CONTROLLER_NAME_LEGACY = 'AdminFeatures';
+    public const POSITION_IDENTIFIER = 'id_feature';
+    public const TABLE = 'feature';
+    public const LIST_ID = 'feature';
+    public const CLASS_NAME = 'Feature';
+    public const IDENTIFIER = 'id_feature';
 
-            if (null !== $handlerResult->getIdentifiableObjectId()) {
-                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
-
-                //@todo change route to index when it's migrated
-                return $this->redirectToRoute('admin_features_create');
-            }
-        } catch (Exception $e) {
-            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
-        }
+    /**
+     * @var string
+     */
+    public $php_self;
 
-        return $this->render('@PrestaShop/Admin/Sell/Catalog/Features/create.html.twig', [
-            'featureForm' => $featureForm->createView(),
-        ]);
-    }
+    /**
+     * @var ControllerConfiguration
+     */
+    public $controllerConfiguration;
 
     /**
-     * Edit feature action.
-     *
-     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
-     *
-     * @param int $featureId
-     * @param Request $request
-     *
-     * @return Response
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
      */
-    public function editAction($featureId, Request $request)
+    public function indexAction(Request $request)
     {
-        try {
-            $editableFeature = $this->getQueryBus()->handle(new GetFeatureForEditing((int) $featureId));
-        } catch (Exception $e) {
-            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
-
-            // @todo change route to features index when it's migrated
-            return $this->redirectToRoute('admin_features_create');
-        }
+        $this->setListFields();
+        $this->buildActionList();
+
+        $helperListConfiguration = $this->get('prestashop.core.bridge.helper_list_configuration_factory')->create([
+            'table' => self::TABLE,
+            'listId' => self::LIST_ID,
```

Comment:
```suggestion
            'listId' => static::LIST_ID,
```

Reply:
Sorry, I don't see the difference between `static::` and `self::`? And why do I have to do this? @kpodemski

Reply:
`self` always refers to the class where the key word is used (i.e. the parent class)
`static` refers to the current class if redefined (i.e. the inherited class), or to the parent class otherwise

Here using `static` is a safety measure that allows overriding the constant in inherited class, to get used by methods of the parent class. This is the "late state bindings" behaviour of PHP language:
https://www.php.net/manual/en/language.oop5.late-static-bindings.php